packages/cli/src/site-configs.types.ts

```diff
@@ -0,0 +1,21 @@
+export type SiteConfig = {
+    name: string;
+    contentScope: Record<string, unknown>;
+    domains: {
+        main: string;
+        preliminary?: string;
+        pattern?: string;
```

Comment:
Should we allow `RegExp` here as well?

<img width="342" alt="image" src="https://github.com/vivid-planet/comet/assets/48853629/7cc61ddc-5b4f-4cef-bb2f-a743678b301d">

Reply:
better only RegExp, why would you ever use a string?